x8.md

```diff
@@ -51,28 +48,30 @@ For more details regarding serialization and deserialization please consult [tra
 
 # Message types
 
-- [Message](##Message)
+- [Message](#message)
 
 ## Message
 
 The type `Message` represents a text message exchanged between clients.
 
-- [Payload](###Payload)
-- [Content types](###Content-types)
-- [Message types](###Message-types)
-- [Clock vs Timestamp and message ordering](###Clock-vs-Timestamp-and-message-ordering)
-- [Replies](###Replies)
+- [Payload](#payload)
+- [Content types](#content-types)
+- [Message types](#message-types)
+- [Clock vs Timestamp and message ordering](#clock-vs-timestamp-and-message-ordering)
+- [Replies](#replies)
 
 ### Payload
 
-Payload is a struct (a compound data type) with the following fields (order is importent):
-1. text `string`
-2. content type `enum` (more in [Content types](#content-types))
-3. message type `enum` (more in [Message types](#message-types))
-4. clock `int64`
-5. timestamp `int64`
-6. content `struct { chat-id string, text string }`
+Payload is a struct (a compound data type) with the following fields (order is important):
```

Comment:
Can we be more precise in struct description? A la RFC, e.g. TLS style https://tools.ietf.org/html/rfc8446

Reply:
added as todo inline